### PR TITLE
Document kvstore leases, update links

### PR DESCRIPTION
**Pull request by joestringer**
* Update the `kvstore` code links to pkg.go.dev
* List the objects by lease-ID with lease/GC behaviour in a nice table

### CODE REVIEW

**Review by aanm — APPROVED** [review]

File: Documentation/kvstore.rst [hunk 1]
```diff
@@ -114,7 +114,26 @@ store in the event that an agent dies on a particular and never reappears.
 
 The lease lifetime is set to 15 minutes. The exact expiration behavior is
 dependent on the kvstore implementation but the expiration typically occurs
-after double the lifetime
+after double the lease lifetime.
+
+In addition to regular entry leases, all locks in the key-value stored are
```
Comment (pchaigno) [inline Documentation/kvstore.rst]:
```suggestion
In addition to regular entry leases, all locks in the key-value store are
```

**Review by pchaigno — COMMENTED** [review]
One nit. LGTM otherwise!

File: Documentation/kvstore.rst (hunk 1 above)
Comment (joestringer) [reply Documentation/kvstore.rst]: Good spotting.

**Comment by joestringer** [comment]
Docs check passed, docs-only change. Merging.

**Pull request merged**